File: back/internal/model/journal.go
package model

import "time"

// JournalEntry represents a training session log.
type JournalEntry struct {
	ID              string    `json:"id"`
	UserID          string    `json:"user_id"`
	SessionDate     string    `json:"session_date"`
	DurationMinutes int       `json:"duration_minutes"`
	Intensity       int       `json:"intensity"`
	Notes           string    `json:"notes"`
	CreatedAt       time.Time `json:"created_at"`
}

File: front/types/journal.ts
export type JournalEntry = {
  id: string;
  user_id?: string;
  session_date: string;
  duration_minutes: number;
  intensity: number;
  notes: string;
  created_at: string;
};

export type JournalEntryPayload = {
  session_date: string;
  duration_minutes: number;
  intensity: number;
  notes: string;
};

export type JournalListResponse = {
  data: JournalEntry[];
  count: number;
};

export type JournalSingleResponse = {
  data: JournalEntry;
};
